src/file_pipe_log/pipe.rs

```diff
@@ -183,81 +76,59 @@ impl<F: FileSystem> SinglePipe<F> {
                 alignment = 16;
             }
         }
-
-        let create_file = first_seq == 0;
-        let active_seq = if create_file {
-            first_seq = 1;
-            let file_id = FileId {
-                queue,
-                seq: first_seq,
-            };
-            let fd = Arc::new(file_system.create(&file_id.build_file_path(&cfg.dir))?);
-            fds.push_back(FileWithFormat {
-                handle: fd,
-                format: LogFileFormat::new(cfg.format_version, alignment),
-            });
-            first_seq
-        } else {
-            first_seq + fds.len() as u64 - 1
+        let file_format = LogFileFormat::new(cfg.format_version, alignment);
+        // Update default file format in the file collection.
+        files.upd_file_format(file_format);
```

Comment:
(1) why not pass it in when creating the collection (2) this function name is bad

Reply:
A compromised way to avoid too many arguments for FileCollection::new(...).

I'll use `allow(clippy::too_many_arguments)` here.